api/configuration/v1alpha1/kongkey_types.go

```diff
@@ -0,0 +1,182 @@
+/*
+Copyright 2024 Kong, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	konnectv1alpha1 "github.com/kong/kubernetes-configuration/api/konnect/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// KongKey is the schema for KongKey API which defines a KongKey entity.
+//
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Programmed",description="The Resource is Programmed on Konnect",type=string,JSONPath=`.status.conditions[?(@.type=='Programmed')].status`
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.spec.controlPlaneRef) || has(self.spec.controlPlaneRef)", message="controlPlaneRef is required once set"
```

Comment:
Should we add a validation rule here to disable `controlPlaneRef` to specify `namespace`?

Reply:
@randmonkey You probably mean something like this https://github.com/Kong/kubernetes-configuration/pull/82/files#diff-134719f357e17ee676fb459e368e424fe001e8a36b50ed06128f3faf2d78d5dbR37 from #82? Yes, I believe we should be consistent with enforcing this until we have cross-namespace references designed and implemented.

Reply:
Added the validation: [bda92a0](https://github.com/Kong/kubernetes-configuration/pull/87/commits/bda92a0934edbc3241cfea7e9a26a63cc36adf07)